OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/gpx/GpxUtilities.kt

```diff
@@ -888,19 +916,41 @@ object GpxUtilities {
 		return parseTime(text, getTimeFormatterTZ())
 	}
 
+	private fun preprocessGpxTime(input: String): String {
+		// fix 2005-05-07T05:45:04+04:00Z
+		return if (input.contains("+") || input.contains("-")) {
+			input.removeSuffix("Z")
+		} else {
+			input
+		}
+	}
+
 	private fun parseTime(text: String, format: DateTimeFormat<DateTimeComponents>): Long {
-		var time: Long = 0
-		try {
-			time = flexibleGpxTimeParser(text, format)
-		} catch (e: Exception) {
+		val processedText = preprocessGpxTime(text)
+		val formats = listOf(
+			format,
+			getTimeNoTimeZoneFormatter(),
+			getTimeFormatterTZ(),
```

Comment:
getTimeFormatterTZ() could be excluded because it's already duplicated with:

	fun parseTime(text: String): Long {
		return parseTime(text, getTimeFormatterTZ())
	}